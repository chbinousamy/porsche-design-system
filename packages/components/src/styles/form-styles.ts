import type { JssStyle, Styles } from 'jss';
import type { Theme } from '../types';
import { getThemedColors, getTransition, hoverMediaQuery, prefersColorSchemeDarkMediaQuery } from './';
import {
  borderRadiusSmall,
  borderWidthBase,
  fontLineHeight,
  spacingStaticSmall,
  textSmallStyle,
} from '@porsche-design-system/utilities-v2';
import { getThemedFormStateColors } from './form-state-color-styles';
import type { FormState } from '../utils/form/form-state';

// TODO: should be removed if possible?
export const INPUT_HEIGHT = 54;

export type ChildSelector = 'input' | 'select' | 'textarea';

export const getSlottedInputTextareaSelectStyles = (
  child: ChildSelector,
  state: FormState,
  theme: Theme,
  additionalDefaultJssStyle?: JssStyle
): Styles => {
  const { primaryColor, contrastLowColor, contrastMediumColor, disabledColor } = getThemedColors(theme);
  const {
    primaryColor: primaryColorDark,
    contrastLowColor: contrastLowColorDark,
    contrastMediumColor: contrastMediumColorDark,
    disabledColor: disabledColorDark,
  } = getThemedColors('dark');
  const { formStateColor, formStateHoverColor } = getThemedFormStateColors(theme, state);
  const { formStateColor: formStateColorDark, formStateHoverColor: formStateHoverColorDark } = getThemedFormStateColors(
    'dark',
    state
  );

  return {
    [`::slotted(${child})`]: {
      // TODO: gridArea should be defined by integrating style
      gridArea: '3 / 1 / auto / span 3',
      display: 'block',
      width: '100%',
      height:
        child !== 'textarea'
          ? `calc(${fontLineHeight} + 10px + ${borderWidthBase} * 2 + ${spacingStaticSmall} * 2)` // we need 10px additionally so input height becomes 54px
          : 'auto',
      margin: 0,
      outline: 0,
      WebkitAppearance: 'none', // iOS safari
      appearance: 'none',
      boxSizing: 'border-box',
      border: `${borderWidthBase} solid ${formStateColor || contrastMediumColor}`,
      borderRadius: borderRadiusSmall,
      background: 'transparent',
      font: textSmallStyle.font.replace('ex', 'ex + 6px'), // a minimum line-height is needed for input, otherwise value is scrollable in Chrome, +6px is aligned with how Safari visualize date/time input highlighting
      textIndent: 0,
      color: primaryColor,
      transition: ['color', 'border-color', 'background-color'].map(getTransition).join(), // for smooth transitions between e.g. disabled states
      ...prefersColorSchemeDarkMediaQuery(theme, {
        borderColor: formStateColorDark || contrastMediumColorDark,
        color: primaryColorDark,
      }),
      ...additionalDefaultJssStyle,
    },
    ...(hoverMediaQuery({
      // with the media query the selector has higher priority and overrides disabled styles
      [`::slotted(${child}:not(:disabled):not(:focus):not([readonly]):hover),.label:hover~* ::slotted(${child}:not(:disabled):not(:focus):not([readonly]))`]:
        {
          borderColor: formStateHoverColor || primaryColor,
          ...prefersColorSchemeDarkMediaQuery(theme, {
            borderColor: formStateHoverColorDark || primaryColorDark,
          }),
        },
    }) as Styles),
    [`::slotted(${child}:focus)`]: {
      borderColor: primaryColor,
      ...prefersColorSchemeDarkMediaQuery(theme, {
        borderColor: primaryColorDark,
      }),
    },
    [`::slotted(${child}:disabled)`]: {
      cursor: 'not-allowed',
      color: disabledColor,
      borderColor: disabledColor,
      WebkitTextFillColor: disabledColor,
      ...prefersColorSchemeDarkMediaQuery(theme, {
        color: disabledColorDark,
        borderColor: disabledColorDark,
        WebkitTextFillColor: disabledColorDark,
      }),
    },
    ...(child !== 'select' && {
      [`::slotted(${child}[readonly])`]: {
        borderColor: contrastLowColor,
        background: contrastLowColor,
        ...prefersColorSchemeDarkMediaQuery(theme, {
          borderColor: contrastLowColorDark,
          background: contrastLowColorDark,
        }),
      },
    }),
  };
};

export const getUnitCounterStyles = (isDisabled: boolean, theme: Theme): JssStyle => {
  const { disabledColor, contrastMediumColor } = getThemedColors(theme);
  const { disabledColor: disabledColorDark, contrastMediumColor: contrastMediumColorDark } = getThemedColors('dark');

  return {
    pointerEvents: 'none',
    font: textSmallStyle.font,
    color: contrastMediumColor,
    ...prefersColorSchemeDarkMediaQuery(theme, {
      color: contrastMediumColorDark,
    }),
    ...(isDisabled && {
      color: disabledColor,
      cursor: 'not-allowed',
      ...prefersColorSchemeDarkMediaQuery(theme, {
        color: disabledColorDark,
      }),
    }),
  };
};
