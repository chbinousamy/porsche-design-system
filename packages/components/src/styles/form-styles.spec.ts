import { getSlottedTextFieldTextareaSelectStyles, getUnitCounterStyles } from './form-styles';

describe('getSlottedTextFieldTextareaSelectStyles()', () => {
  it.each<Parameters<typeof getSlottedTextFieldTextareaSelectStyles>>([
    ['input', 'none', false, 'light', undefined],
    ['input', 'none', true, 'light', undefined],
    ['input', 'success', false, 'light', undefined],
    ['input', 'error', false, 'light', undefined],
    ['select', 'none', false, 'dark', undefined],
    ['select', 'success', false, 'dark', undefined],
    ['select', 'error', false, 'dark', undefined],
    ['textarea', 'none', false, 'dark', undefined],
    ['textarea', 'success', false, 'dark', undefined],
    ['textarea', 'error', false, 'dark', undefined],
    ['textarea', 'error', false, 'dark', { bottom: 5 }],
  ])(
    'should return correct css for child: %s, state: %s, isLoading: %s, theme: %s and additionalDefaultJssStyle: %o',
    (...args) => {
      expect(getSlottedTextFieldTextareaSelectStyles(...args)).toMatchSnapshot();
    }
  );
});

describe('getUnitCounterStyles()', () => {
  it.each<Parameters<typeof getUnitCounterStyles>>([
    [false, 'light'],
    [true, 'light'],
    [false, 'dark'],
    [true, 'dark'],
    [false, 'auto'],
    [true, 'auto'],
  ])('should return correct css for isDisabled: %s and theme: %s', (...args) => {
    expect(getUnitCounterStyles(...args)).toMatchSnapshot();
  });
});
