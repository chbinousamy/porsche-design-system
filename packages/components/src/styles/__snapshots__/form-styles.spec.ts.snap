// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: input, state: error, theme: light, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(input)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #CC1922",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#010205",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(input:disabled)": {
    "WebkitTextFillColor": "#949598",
    "borderColor": "#949598",
    "color": "#949598",
    "cursor": "not-allowed",
  },
  "::slotted(input:focus)": {
    "borderColor": "#010205",
  },
  "::slotted(input[readonly])": {
    "background": "#D8D8DB",
    "borderColor": "#D8D8DB",
  },
  "@media(hover:hover)": {
    "::slotted(input:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(input:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#951219",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: input, state: none, theme: light, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(input)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #6B6D70",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#010205",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(input:disabled)": {
    "WebkitTextFillColor": "#949598",
    "borderColor": "#949598",
    "color": "#949598",
    "cursor": "not-allowed",
  },
  "::slotted(input:focus)": {
    "borderColor": "#010205",
  },
  "::slotted(input[readonly])": {
    "background": "#D8D8DB",
    "borderColor": "#D8D8DB",
  },
  "@media(hover:hover)": {
    "::slotted(input:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(input:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#010205",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: input, state: none, theme: light, isLoading: true and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(input)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #6B6D70",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#010205",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(input:disabled)": {
    "WebkitTextFillColor": "#949598",
    "borderColor": "#949598",
    "color": "#949598",
    "cursor": "not-allowed",
  },
  "::slotted(input:focus)": {
    "borderColor": "#010205",
  },
  "::slotted(input[readonly])": {
    "background": "#D8D8DB",
    "borderColor": "#D8D8DB",
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: input, state: success, theme: light, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(input)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #197E10",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#010205",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(input:disabled)": {
    "WebkitTextFillColor": "#949598",
    "borderColor": "#949598",
    "color": "#949598",
    "cursor": "not-allowed",
  },
  "::slotted(input:focus)": {
    "borderColor": "#010205",
  },
  "::slotted(input[readonly])": {
    "background": "#D8D8DB",
    "borderColor": "#D8D8DB",
  },
  "@media(hover:hover)": {
    "::slotted(input:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(input:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#0E4809",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: select, state: error, theme: dark, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(select)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #FC4040",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(select:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(select:focus)": {
    "borderColor": "#FBFCFF",
  },
  "@media(hover:hover)": {
    "::slotted(select:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(select:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#FB0404",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: select, state: none, theme: dark, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(select)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #88898C",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(select:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(select:focus)": {
    "borderColor": "#FBFCFF",
  },
  "@media(hover:hover)": {
    "::slotted(select:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(select:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#FBFCFF",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: select, state: success, theme: dark, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(select)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #09D087",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2)",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(select:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(select:focus)": {
    "borderColor": "#FBFCFF",
  },
  "@media(hover:hover)": {
    "::slotted(select:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(select:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#069561",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: textarea, state: error, theme: dark, isLoading: false and additionalDefaultJssStyle: { bottom: 5 } 1`] = `
{
  "::slotted(textarea)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #FC4040",
    "borderRadius": "4px",
    "bottom": 5,
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "auto",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(textarea:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(textarea:focus)": {
    "borderColor": "#FBFCFF",
  },
  "::slotted(textarea[readonly])": {
    "background": "#404044",
    "borderColor": "#404044",
  },
  "@media(hover:hover)": {
    "::slotted(textarea:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(textarea:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#FB0404",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: textarea, state: error, theme: dark, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(textarea)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #FC4040",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "auto",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(textarea:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(textarea:focus)": {
    "borderColor": "#FBFCFF",
  },
  "::slotted(textarea[readonly])": {
    "background": "#404044",
    "borderColor": "#404044",
  },
  "@media(hover:hover)": {
    "::slotted(textarea:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(textarea:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#FB0404",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: textarea, state: none, theme: dark, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(textarea)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #88898C",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "auto",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(textarea:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(textarea:focus)": {
    "borderColor": "#FBFCFF",
  },
  "::slotted(textarea[readonly])": {
    "background": "#404044",
    "borderColor": "#404044",
  },
  "@media(hover:hover)": {
    "::slotted(textarea:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(textarea:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#FBFCFF",
    },
  },
}
`;

exports[`getSlottedTextFieldTextareaSelectStyles() should return correct css for child: textarea, state: success, theme: dark, isLoading: false and additionalDefaultJssStyle: undefined 1`] = `
{
  "::slotted(textarea)": {
    "WebkitAppearance": "none",
    "appearance": "none",
    "background": "transparent",
    "border": "2px solid #09D087",
    "borderRadius": "4px",
    "boxSizing": "border-box",
    "color": "#FBFCFF",
    "display": "block",
    "font": "normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "height": "auto",
    "margin": 0,
    "outline": 0,
    "textIndent": 0,
    "transition": "color var(--p-transition-duration, .24s) ease,border-color var(--p-transition-duration, .24s) ease,background-color var(--p-transition-duration, .24s) ease",
    "width": "100%",
  },
  "::slotted(textarea:disabled)": {
    "WebkitTextFillColor": "#7E7F82",
    "borderColor": "#7E7F82",
    "color": "#7E7F82",
    "cursor": "not-allowed",
  },
  "::slotted(textarea:focus)": {
    "borderColor": "#FBFCFF",
  },
  "::slotted(textarea[readonly])": {
    "background": "#404044",
    "borderColor": "#404044",
  },
  "@media(hover:hover)": {
    "::slotted(textarea:not(:disabled):not(:focus):not([readonly]):hover),label:hover~* ::slotted(textarea:not(:disabled):not(:focus):not([readonly])),label:hover~* ::part(select-wrapper-dropdown)": {
      "borderColor": "#069561",
    },
  },
}
`;

exports[`getUnitCounterStyles() should return correct css for isDisabled: false and theme: auto 1`] = `
{
  "@media (prefers-color-scheme: dark)": {
    "color": "#88898C",
  },
  "boxSizing": "border-box",
  "color": "#6B6D70",
  "font": "normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "maxWidth": "100%",
  "overflow": "hidden",
  "pointerEvents": "none",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`getUnitCounterStyles() should return correct css for isDisabled: false and theme: dark 1`] = `
{
  "boxSizing": "border-box",
  "color": "#88898C",
  "font": "normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "maxWidth": "100%",
  "overflow": "hidden",
  "pointerEvents": "none",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`getUnitCounterStyles() should return correct css for isDisabled: false and theme: light 1`] = `
{
  "boxSizing": "border-box",
  "color": "#6B6D70",
  "font": "normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "maxWidth": "100%",
  "overflow": "hidden",
  "pointerEvents": "none",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`getUnitCounterStyles() should return correct css for isDisabled: true and theme: auto 1`] = `
{
  "@media (prefers-color-scheme: dark)": {
    "color": "#7E7F82",
  },
  "boxSizing": "border-box",
  "color": "#949598",
  "font": "normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "maxWidth": "100%",
  "overflow": "hidden",
  "pointerEvents": "none",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`getUnitCounterStyles() should return correct css for isDisabled: true and theme: dark 1`] = `
{
  "boxSizing": "border-box",
  "color": "#7E7F82",
  "font": "normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "maxWidth": "100%",
  "overflow": "hidden",
  "pointerEvents": "none",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`getUnitCounterStyles() should return correct css for isDisabled: true and theme: light 1`] = `
{
  "boxSizing": "border-box",
  "color": "#949598",
  "font": "normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "maxWidth": "100%",
  "overflow": "hidden",
  "pointerEvents": "none",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;
